src/main/java/org/systemli/keycloak/UserliUserStorageProviderFactory.java

```diff
@@ -1,48 +1,50 @@
 package org.systemli.keycloak;
 
-import java.util.List;
-
 import org.keycloak.component.ComponentModel;
+import org.keycloak.component.ComponentValidationException;
 import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
 import org.keycloak.provider.ProviderConfigProperty;
 import org.keycloak.provider.ProviderConfigurationBuilder;
 import org.keycloak.storage.UserStorageProviderFactory;
+import org.keycloak.utils.StringUtil;
+
+import java.util.List;
 
 public class UserliUserStorageProviderFactory implements UserStorageProviderFactory<UserliUserStorageProvider> {
-    public static final String PROVIDER_ID = "userli-storage-provider";
-
-    protected final List<ProviderConfigProperty> configMetadata;
-
-    public UserliUserStorageProviderFactory() {
-        configMetadata = ProviderConfigurationBuilder.create()
-                .property()
-                .name("domain")
-                .label("Domain")
-                .type(ProviderConfigProperty.STRING_TYPE)
-                .defaultValue("systemli.org")
-                .helpText("Domain which is allowed to login")
-                .required(true)
-                .add()
-                .build();
-    }
-
-    @Override
-    public UserliUserStorageProvider create(KeycloakSession session, ComponentModel model) {
-        return new UserliUserStorageProvider(session, model);
-    }
-
-    @Override
-    public String getId() {
-        return PROVIDER_ID;
-    }
-
-    @Override
-    public String getHelpText() {
-        return "Userli User Storage Provider";
-    }
-
-    @Override
-    public List<ProviderConfigProperty> getConfigProperties() {
-        return configMetadata;
-    }
+
+        public static final String PROVIDER_ID = "userli-api-user-provider";
+
+        @Override
+        public UserliUserStorageProvider create(KeycloakSession session, ComponentModel model) {
+                return new UserliUserStorageProvider(session, model);
+        }
+
+        @Override
+        public String getId() {
+                return PROVIDER_ID;
+        }
+
+        @Override
+        public String getHelpText() {
+                return "Userli API User Provider";
+        }
+
+        @Override
+        public List<ProviderConfigProperty> getConfigProperties() {
+                return ProviderConfigurationBuilder.create()
+                        .property(Constants.REALM_DOMAIN, "Realm domain", "Limit users to this domain", ProviderConfigProperty.STRING_TYPE, "", null)
+                        .property(Constants.BASE_URL, "Base URL", "Base URL of the API", ProviderConfigProperty.STRING_TYPE, "", null)
+                        .property(Constants.KEYCLOAK_API_TOKEN, "Keycloak API token", "Token for the userli keycloak API", ProviderConfigProperty.STRING_TYPE, "", null)
```

Comment:
```suggestion
                        .property(Constants.API_TOKEN, "API token", "Token for the userli keycloak API", ProviderConfigProperty.STRING_TYPE, "", null)
```

Would call that only API token.